src/test/java/org/mifos/integrationtest/cucumber/stepdef/ReportManagementSteps.java

```diff
@@ -0,0 +1,312 @@
+            JsonNode jsonResponse = objectMapper.readTree(scenarioScopeState.response);
+
+            MatcherAssert.assertThat(jsonResponse.get("id").asText(), Matchers.equalTo(scenarioScopeState.reportId));
+
+            MatcherAssert.assertThat(jsonResponse.get("reportName").asText(), Matchers.equalTo("Updated Report Name"));
+            MatcherAssert.assertThat(jsonResponse.get("description").asText(), Matchers.equalTo("Updated description"));
+
+        } catch (Exception e) {
+            // Handle any JSON parsing exceptions
+            logger.error("Error parsing JSON response: {}", e.getMessage());
+            fail("Error parsing JSON response");
+        }
+    }
+
+    @When("I call the get single report API with expected status of {int}")
+    public void iCallGetSingleReportAPI(int expectedStatus) {
+        try {
+            RequestSpecification requestSpec = Utils.getDefaultSpec();
+
+            String reportId = scenarioScopeState.reportId;
+            String getSingleReportEndpoint = "/reports/{reportId}";
+
+            scenarioScopeState.response = RestAssured.given(requestSpec).header("Platform-TenantId", scenarioScopeState.tenant)
+                    .baseUri(operationsAppConfig.operationAppContactPoint).pathParam("reportId", reportId) 
+                    .expect().statusCode(expectedStatus).when().get(getSingleReportEndpoint).andReturn().asString();
+
+            logger.info("Get Single Report Response: {}", scenarioScopeState.response);
+        } catch (Exception e) {
+            logger.error("Error calling get single report API: {}", e.getMessage());
+            fail("Error calling get single report API");
+        }
+    }
+
+    @Then("the response should contain the details of the requested report")
+    public void theResponseShouldContainRequestedReportDetails() {
+        try {
+            ObjectMapper objectMapper = new ObjectMapper();
+            JsonNode jsonResponse = objectMapper.readTree(scenarioScopeState.response);
+
+            MatcherAssert.assertThat(jsonResponse.get("id").asText(), Matchers.equalTo(scenarioScopeState.reportId));
+            MatcherAssert.assertThat(jsonResponse.get("reportName").asText(), Matchers.equalTo("Updated Report Name"));
+            MatcherAssert.assertThat(jsonResponse.get("description").asText(), Matchers.equalTo("Updated description"));
+
+        } catch (Exception e) {
+            logger.error("Error parsing JSON response: {}", e.getMessage());
+            fail("Error parsing JSON response");
+        }
+    }
+
+    @When("I call the create report API with invalid data")
+    public void iCallCreateReportAPIWithInvalidData() {
+        ObjectMapper objectMapper = new ObjectMapper();
+        ReportRequestDTO reportDTO = new ReportRequestDTO();
+
+        reportDTO.setReportName(null);
+        reportDTO.setReportType("InvalidType");
+        reportDTO.setReportSubType("InvalidSubType");
+        reportDTO.setReportCategory("Operational");
```

Comment:
How are these values other than report Name Invalid, what checks would they fail? There are no checks for these fields in your API.